0000-py-version-support.md

```diff
@@ -0,0 +1,244 @@
+with Python's End of Life schedule.
+
+# Motivation
+[motivation]: #motivation
+
+Documentation and Communication
+-------------------------------
+One of the motivating factors around this policy change is to make it more clear what our Python
+version support is for each Salt releases. If it is clearly defined, then packagers, community members,
+and Salt users will know when a Python version will be dropped for a given Salt release.
+In the past we had never clearly defined how our Python version support lined up with Python's
+End of Life schedule. We previously made a best effort to support the Python versions that lined
+up with the operating systems Python versions, but never clearly defined this and did not always
+follow this in every case. For example, when we dropped Python 2.6 support we still supported
+Redhat 6, which had Python 2.6 installed by default. To continue supporting Redhat 6 while also
+removing support for the End of Life Python 2.6 version, we included Python 2.7 in our packages
+for that operating system.
+
+Including Python in Packages
+----------------------------
+Another motivating factor to make this policy change is our shift to including Python in our
+packages. For the Sodium release we will be using salt-bin (https://github.com/saltstack/salt-bin) to
+package for our Fully Supported Platforms.
+(http://get.saltstack.com/rs/304-PHQ-615/images/SaltStack-Supported-Operating-Systems.pdf)
+This will allow us to easily continue supporting operating systems that include End of Life Python
+versions by default by providing the Python binary in the Salt package included in a Salt release.
+
+Security and Maintenance Concerns
+---------------------------------
+The last motivating factor surrounds security and maintenance concerns. Continuing to support an End
+of Life Python version adds additional maintenance around testing and development for each version.
+Supporting these versions also puts users at a security risk, since no more security fixes will be
+submited and released with that Python version if the operating system does not choose to update
+it themselves.
+
+# Design
+[design]: #detailed-design
+
+The design goal of this SEP is to define a clear policy around Salt's Python version support.
+This new policy defines only supporting major Python versions until they reach End of Life. You can see
+the current status of supported Python branches here https://devguide.Python.org/#status-of-Python-branches
+with the associated End Of Life date for each branch. You can also see details about Python versions that
+have already become End Of Life here: https://devguide.Python.org/devcycle/#end-of-life-branches.
+
+What if a Python version drops support relatively close to a Salt Release?
+--------------------------------------------------------------------------
+We may drop support for a Python version if the End of Life date lands relatively close to a Salt release.
```

Comment:
Dropping support for a Python version means that developers can now use new features available since the oldest supported version. If the release is close to a EOL of a Python version, there will be no time to add new features (and do mostly QA)

So I wonder if makes sense to still support this Python version, or communicate the drop very early on the release process.

Reply:
Yeah it will need to be clearly communicated and decided beforehand what release it will be dropped in particularly for contributors. We will need to communicate the deprecation decision at the beginning of a feature release development cycle.

Reply:
i'll add a clarification note here around that.